### ISSUE TITLE
Attribute error even with "from plantcv import plantcv as pcv"

### ISSUE THREAD

**Issue by scsolano** [danforthcenter/plantcv#363]
### Description

Hello all!

 I am trying to use PlantCV on Jupyter. However when I try to use a function (in this case, **readimage** used in the script as **img, path, filename = pcv.readimage(rgb_img)**), it throws a Attribute Error at me:

"AttributeError: module 'plantcv.plantcv' has no attribute 'readimage'"

 Instead of "**import plantcv**", I used "**from plantcv import plantcv as pcv**", with and without "as pcv" as suggested in #246, but I still have an attribute error. 

What would you guys suggest? 

 ### Details

If it helps, I am using the latest update of Anaconda for Jupyter, Python 3.7.3 and the latest version of PlantCV. Thanks!

**Comment by nfahlgren** [comment]
Hi @scsolano! My best guess is that in the location you started Jupyter from there is a plantcv folder. Python is recognizing that folder as the plantcv namespace rather than looking for the installed version. If that's the case and you either move the plantcv folder elsewhere or start Jupyter from another location I think that will fix it. But let us know if the issue persists!

**Comment by scsolano** [comment]
Hello, thanks for the response!

I start Jupyter from Anaconda Navigator (so I am not sure where a plantcv folder would be in Navigator), so instead of starting with Navigator I just started from Windows Start and it still gives me an attribute error.

The scripts I try to run in Jupyter are in the same place as the plantcv folder if that is what you are referring to though? 

![screenshot](https://user-images.githubusercontent.com/49666313/56182467-6434b680-5fe0-11e9-9c6f-c5eddbd6de09.PNG)

And in that case, this is what is inside of the plantcv folder

![screenshot2](https://user-images.githubusercontent.com/49666313/56182479-71ea3c00-5fe0-11e9-81a2-e5783b4c651c.PNG)

![screenshot3](https://user-images.githubusercontent.com/49666313/56182483-757dc300-5fe0-11e9-8a1a-8d9425104748.PNG)

![screenshot4](https://user-images.githubusercontent.com/49666313/56182491-77478680-5fe0-11e9-90af-b1eeb411e514.PNG)

**Comment by nfahlgren** [comment]
That’s right, I was referring to your Jupyter scripts being in the same place as the plantcv folder. If you move or rename that plantcv folder, or move your scripts to another place it should work.

If we think about the import as walking through folders, `from plantcv import plantcv` refers to a folder structure of plantcv/plantcv, which are the two folders nested in the GitHub repository. But the repository also is a folder called plantcv, so when Python sees it in the local directory it grabs it as the first `from plantcv`, which  makes it one directory off. If you ran the install command you will want it to import the package from the Python library rather than directly from the source files anyhow. And changing the name of the GitHub directory won’t hurt anything, it’s just a container for the code.

**Comment by scsolano** [comment]
I did rename that folder but it still didn't work. However when I moved the contents of plantcv/plantcv/plantcv (and I renamed the original "plantcv" to something else) to another location, it gives me a new error (ModuleNotFoundError: No module named 'plotnine')

Here is the new location of the plantcv folder that I moved.

![SCR2](https://user-images.githubusercontent.com/49666313/56183584-1bcbc780-5fe5-11e9-9dcb-e5d34115b37b.PNG)

Here is the error.

![SCR3](https://user-images.githubusercontent.com/49666313/56183658-59c8eb80-5fe5-11e9-8b31-6aa60f46289b.PNG)

So it seems like something changed more or less

**Comment by nfahlgren** [comment]
When you make a new Jupyter notebook is there a notebook kernel option with the name of your PlantCV environment? It would be something like `Python [conda env:plantcv]`

**Comment by scsolano** [comment]
Is this what you mean (apologies, I'm new to using Jupyter)?

![SCR4](https://user-images.githubusercontent.com/49666313/56184651-ddd0a280-5fe8-11e9-8f79-f31af60a8881.PNG)

**Comment by HaleySchuhl** [comment]
It looks like your PlantCV environment isn't an option right now in your Jupyter notebook. 

I used the conda tab to create my own environment. This ought to add the kernel you'll need to use. I've found this more straightforward than using the Jupyter terminal. 

![image](https://user-images.githubusercontent.com/44006936/56212707-2d7e9080-6020-11e9-8b8c-a3bb3165f2e7.png)

Here's the site that screen shot is from: https://docs.anaconda.com/ae-notebooks/user-guide/adv-tasks/work-with-environments/#create-default-env-jn

**Comment by scsolano** [comment]
Hello, apologies for not responding quickly I've been quite busy - yes it worked! It seems like I wasn't opening Jupyter within the PlantCV environment but rather the base/root environment. Now it works just fine, thank you all, I'll let you know if I need any further help!

**Comment by nfahlgren** [comment]
Excellent! We should add something about this to the documentation. I have been launching from the command-line and I wasn't aware of how to do this through Anaconda Navigator, great to know!
